symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/message/util/MessageParser.java

```diff
@@ -0,0 +1,132 @@
+package com.symphony.bdk.core.service.message.util;
+
+import com.symphony.bdk.core.service.message.exception.MessageParserException;
+import com.symphony.bdk.gen.api.model.V4Message;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import org.apiguardian.api.API;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Helper class for extracting entities inside an incoming {@link com.symphony.bdk.gen.api.model.V4Message} such as:
+ * Mentions, Hashtags, Cashtags, Emojis.
+ */
+@API(status = API.Status.EXPERIMENTAL)
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public final class MessageParser {
+
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  /**
+   * Parse data inside the message and returns a list containing the user ids corresponding to the users mentioned
+   *
+   * @param message incoming V4 message to be parsed
+   * @return list of users ids that has been mentioned inside the message
+   */
+  public static List<Long> getMentions(V4Message message) throws MessageParserException {
+    List<String> mentionsList = getTags(message, EntityTypeEnum.MENTION);
+    if (mentionsList.isEmpty()) { return Collections.emptyList(); }
```

Comment:
we can remove the isempty check, the stream will work on an empty list, returning an empty list

Reply:
uh right, I forgot to remove it.